syntax = "proto3";

package proxy_db;

enum Status {
  OK = 0;
  FAILED = 1;
}

service ProxyDatabaseCommunication{
  rpc ChangeLeader(LeaderPackage) returns (Confirmation);
  rpc SendLogEntry(LogEntry) returns (Confirmation);
  rpc ReadData(ReadRequest) returns (ReadResponse);
}

message LeaderPackage{
  string leaderIdentifier = 1;
}

message LogEntry{
  string log = 1;
}

message Confirmation{
  Status status = 1;
  string error = 2;
}

message ReadRequest{
  uint64 id = 1;
}

message ReadResponse{
  Status status = 1;
  string data = 2;
  string error = 3;
}
